File: ios/BiangBiang Hanzi/Views/CopyToast.swift
//
//  CopyToast.swift
//  BiangBiang Hanzi
//
//  Brief confirmation toast shown after copying recognized text.
//

import SwiftUI

struct CopyToast: View {
    var body: some View {
        Text("Text copied")
            .font(.subheadline.weight(.semibold))
            .padding(.horizontal, 12)
            .padding(.vertical, 8)
            .background(.ultraThinMaterial)
            .clipShape(.rect(cornerRadius: AppDesign.cornerRadius))
            .shadow(radius: 6)
    }
}

#Preview {
    CopyToast()
}

File: android/app/src/main/java/dev/veeso/harakatlens/ui/components/CopyToast.kt
package dev.veeso.harakatlens.ui.components

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.slideInVertically
import androidx.compose.animation.slideOutVertically
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import dev.veeso.harakatlens.ui.AppDesign

/**
 * Brief confirmation toast shown after copying recognized text.
 */
@Composable
fun CopyToast(visible: Boolean, message: String = "Text copied") {
    AnimatedVisibility(
        visible = visible,
        enter = fadeIn() + slideInVertically { it / 2 },
        exit = fadeOut() + slideOutVertically { it / 2 },
    ) {
        Text(
            text = message,
            fontWeight = FontWeight.SemiBold,
            style = MaterialTheme.typography.bodyMedium,
            modifier = Modifier
                .shadow(6.dp, RoundedCornerShape(AppDesign.cornerRadius))
                .clip(RoundedCornerShape(AppDesign.cornerRadius))
                .background(MaterialTheme.colorScheme.surfaceContainerHigh)
                .padding(horizontal = 12.dp, vertical = 8.dp),
        )
    }
}
